src/wms/src/main/java/org/geoserver/wms/legendgraphic/GetLegendGraphicKvpReader.java

```diff
@@ -214,6 +227,50 @@ public GetLegendGraphicRequest read(Object req, Map kvp, Map rawKvp) throws Exce
         return request;
     }
 
+    /**
+     * Create a Legend request to fetch Legend from cascaded WMS service
+     *
+     * @param layerInfo of Cascaded Layer, should WMSLayerInfo
+     * @param request should be instance of GetLegendGraphicRequest
+     * @return GetCascadedLegendGraphicRequest
+     * @throws IOException
+     */
+    private LegendRequest getCascadeLegendRequest(
+            LayerInfo layerInfo, GetLegendGraphicRequest request) throws IOException {
+        WMSLayerInfo wmsLayerInfo = (WMSLayerInfo) layerInfo.getResource();
+        WMSStoreInfo wmsStoreInfo = wmsLayerInfo.getStore();
+        WebMapServer wmsServer = wmsStoreInfo.getWebMapServer(null);
+
+        org.geotools.ows.wms.request.GetLegendGraphicRequest remoteLegendGraphicRequest;
+
+        if (wmsServer.getCapabilities().getVersion().equalsIgnoreCase("1.3.0")) {
+            // WebMapServer under 1.3.0 version does not create GetLegendGraphicRequest
+            // since the XML tag <GetLegendGraphic> is not present under the <Capability>
+            // HACK HACK HACK
```

Comment:
?

Reply:
here we are checking if the WMS Server was created using 1.3.0 capabilities document, because that will fail `wmsServer.createGetLegendGraphicRequest()`. So we take a valid empty GetMap request and turn it into GetLegend request